Formula/helix.rb

```diff
@@ -19,7 +26,29 @@ class Helix < Formula
   end
 
   def install
-    libexec.install Dir["*"]
-    (bin/"hx").write_env_script(libexec/"hx", :HELIX_RUNTIME => libexec/"runtime")
+    if build.head?
+      system "cargo", "install", *std_cargo_args(path: "helix-term")
+      libexec.install "runtime"
+      libexec.install "target/release/hx"
+    else
+      libexec.install Dir["*"]
+    end
+
+    (bin / "hx").write_env_script(libexec / "hx", HELIX_RUNTIME: libexec / "runtime")
+  end
+
+  def caveats
+    if build.head?
+      <<~EOS
+        Building tree-sitter grammars:
+
+        Tree-sitter grammars must be fetched and compiled if not pre-packaged.
+        Fetch grammars with `hx --grammar fetch` (requires git) and compile them with `hx --grammar build` (requires a C compiler).
+      EOS
+    end
+  end
```

Comment:
This might be unnecessary: `cargo install` fetches and builds grammars in a build script unless a `HELIX_DISABLE_AUTO_GRAMMAR_BUILD` env var is set.

Reply:
> This might be unnecessary: `cargo install` fetches and builds grammars in a build script unless a `HELIX_DISABLE_AUTO_GRAMMAR_BUILD` env var is set.

I guess we can remove it, but it's a step from the `build from sources` doc from here: https://github.com/helix-editor/helix#installation, 🤔

Reply:
> Tree-sitter grammars must be fetched and compiled if not pre-packaged.

This build definition is packaging the tree-sitter grammars because it doesn't set `HELIX_DISABLE_AUTO_GRAMMAR_BUILD`. That note is geared towards installations from a package manager that might explicitly disable the grammar build (for example if the package manager has the grammars as separate packages, or if the package manager disables network access during builds).